src/components/forms/Search/index.tsx

```diff
@@ -1,71 +1,84 @@
-import { History, Location } from 'history';
-import React, { useEffect, useState } from 'react';
-import { Button, Form, FormGroup, Input } from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { debounce } from 'lodash';
+import queryString from 'querystring';
+import React from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
+import { Input } from 'reactstrap';
 import { SEARCH } from '../../../configs/lang';
 import { QUERY_PARAM_TITLE } from '../../../constants';
 import { getQueryParams } from '../../../helpers/utils';
+import './index.css';
 
-/**
- * Interface for handleSearchChange event handler
- */
-export type Change = (event: React.ChangeEvent<HTMLInputElement>) => void;
+/** call handler function after this many milliseconds since when it was last invoked */
+export const DEBOUNCE_HANDLER_MS = 1000;
 
-/**
- * Interface for handleSubmit event handler
- */
-export type Submit = (event: React.FormEvent<HTMLFormElement>) => void;
+/** function type for custom onChangeHandler functions */
+export type OnChangeType = (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
I thought the Search updates were happening in other PRs, is this accounting for those changes? Is it iterating on those changes?

Reply:
I built on the work that Kelvin had done for this. where by instead of clicking on a search, one know only types, on the input element and the plans are searched automatically.

Reply:
cool sounds good